server/commands/subscriptionCreator.ts

```diff
@@ -0,0 +1,59 @@
+import { Transaction } from "sequelize";
+import { Subscription, Event, User } from "@server/models";
+
+type Props = {
+  /** The user creating the subscription */
+  user: User;
+  /** The document to subscribe to */
+  documentId?: string;
+  /** Event to subscribe */
+  event: string;
+  /**  Fetch soft deleted rows? */
+  paranoid?: boolean;
+  /** The IP address of the user creating the subscription */
+  ip: string;
+  transaction: Transaction;
+};
+
+/**
+ * This command creates a "subscription" on
+ * a document via the subscription relation.
+ *
+ * @param Props The properties of the subscription to create
+ * @returns Subscription The subscription that was created
+ */
+export default async function subscriptionCreator({
+  user,
+  documentId,
+  event,
+  paranoid = true,
+  ip,
+  transaction,
+}: Props): Promise<Subscription> {
+  const [subscription, created] = await Subscription.findOrCreate({
+    where: {
+      userId: user.id,
+      documentId,
+      event,
+    },
+    paranoid,
```

Comment:
```suggestion
    paranoid: false,
```

This find should always find previously deleted subscriptions